charts/karmada/templates/_helpers.tpl

```diff
@@ -631,9 +622,18 @@ kubectl config use-context karmada-host-context
     - -c
     - |
       bash <<'EOF'
-      {{- include "karmada.initContainer.build-kubeconfig" . | nindent 6 }}
-      kubectl wait --for=condition=complete job {{ include "karmada.name" . }}-static-resource -n {{ include "karmada.namespace" . }}
+      set -ex
+
+      # here are three cases:
+      # case first installation: no `cm/karmada-versions` at first, so when you get it, it means `karmada-static-resource-job` finished.
+      # case restart: already has `cm/karmada-versions`, which means `karmada-static-resource-job` already finished.
+      # case upgrading: already has `cm/karmada-versions`, but it may be old version, we should wait until `.data.karmadaImageVersion` euqal to current `.Values.karmadaImageVersion`.
```

Comment:
```suggestion
      # case upgrading: already has `cm/karmada-versions`, but it may be old version, we should wait until `.data.karmadaImageVersion` equal to current `.Values.karmadaImageVersion`.
```